jax/numpy/lax_numpy.py

```diff
@@ -2648,6 +2648,15 @@ def sort(a, axis=-1, kind='quicksort', order=None):
     return lax.sort(a, _canonicalize_axis(axis, ndim(a)))
 
 
+@_wraps(onp.lexsort)
+def lexsort(keys, axis=-1):
```

Comment:
Unfortunately the problem with this approach is that it won't work under jit transformation, and ideally we'd like that to be true in `jax.numpy` wherever possible. The reason it won't work under jit is that the control flow of sorted depends on the values of a, but they aren't known as concrete values at trace time.

I think, however that `lexsort` can be implemented using the underlying XLA sort operator.
https://www.tensorflow.org/xla/operation_semantics#sort
although the full generality of that operator is not exposed in JAX's `lax` library, which wraps it, so we'd first have to start by generalizing `lax.sort`. (I have a half-finished branch in which I did that myself.)

The same problem is true of `sort_complex` below; to implement `sort_complex`, you'd most likely need to use a tuple-sort of key-value pairs, which is something that XLA supports that is not plumbed into `lax` yet.

My main concern is that this project might make for a difficult starter project!